compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1066,6 +1185,8 @@ impl<'context> Elaborator<'context> {
     }
 
     fn define_type_alias(&mut self, alias_id: TypeAliasId, alias: UnresolvedTypeAlias) {
+        self.push_scope();
+
```

Comment:
Yeah it is due to introducing numeric generics into scope inside of `add_generics`. I see your point about only needing the numeric generics in function scope though and will look at re-ordering when I introduce numeric generics into scope as to avoid these extra scopes.

Reply:
I now only introduce numeric generics into scope when elaborating functions.